### PR TITLE
Tonk Attack: Move colour logic from Tonk Tower plugin to Tonk plugin

### PR DESCRIPTION
**Pull request by Steedie**
## What
This PR moves the Tonk Attack colour logic from TonkTower.js to Tonk.js, the item plugin.

## Why
1. The colour logic requires `alwaysActive: true` which is causing a large number of extra calls to the tonk server than is needed for the game.
2. So that when the player gets rid of their Tonk, it stops trying to set their colour, which solves the case where you're joined into a game of Tonk Attack/in the lobby, but you're also playing another game that uses team colours.

## Notes
Since the logic is now attached to the Tonk plugin instead of an `alwaysActive` tower, they won't see who's playing Tonk Attack unless they themselves have a Tonk in their inventory:
image
This could possibly be fixed by also leaving the logic in the Tonk Tower, but accepting that they will have to click the tower to see who's currently playing.

### CODE REVIEW

**Comment by chrisfarms** [comment]
> Since the logic is now attached to the Tonk plugin instead of an alwaysActive tower, they won't see who's playing Tonk Attack unless they themselves have a Tonk

I think this is fine ... keeps it simple 👍

File: contracts/src/maps/tonk/tonk_kinds/TonkTower/TonkTower.js [hunk 1]
```diff
@@ -538,70 +538,70 @@ async function _update(params) {
...
-                        type: "unit",
-                        key: "color",
-                        id: p.mobile_unit_id,
-                        value: "#2daee0", // BLUE - MAIN
-                    });
-                }
-                break;
-        }
-    });
+    // const mapUnitObj = [];
```
Comment (chrisfarms) [inline contracts/src/maps/tonk/tonk_kinds/TonkTower/TonkTower.js:541]: can we remove all this commented out code? ... can always pull it out of history if we need it

**Comment by Steedie** [comment]
Marked as ready for review

File: contracts/src/maps/tonk/tonk_kinds/TonkTower/Tonk.js [hunk 2]
```diff
@@ -855,6 +855,72 @@ export default async function update(params) {
 
     status = tonkPlayer.eliminated || playerEliminated ? "ELIMINATED" : status;
 
+    // Team colorus
+    const mapUnitObj = [];
+    const { game_status, win_result } = game;
```
[the remark this answers is not in the document]
Comment (Steedie) [reply contracts/src/maps/tonk/tonk_kinds/TonkTower/Tonk.js]: Cheers, I've committed these fixes now 👍

**Review by chrisfarms — APPROVED** [review]

**Pull request merged**